### PR TITLE
JAMES-3534 Identity/set update + delete and Identity/get should fail for identities whose alias is deleted

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/james-project/issues/4433

### CODE REVIEW

File: server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/identity/CustomIdentityDAOContract.scala [hunk 1]
```diff
@@ -156,35 +156,7 @@ trait CustomIdentityDAOContract {
...
-      .containsExactlyInAnyOrder(Identity(id = identity.id,
-        name = IdentityName("Bob (new name)"),
-        email = bob.asMailAddress(),
-        replyTo = Some(List(EmailAddress(Some(EmailerName("My Boss (updated)")), new MailAddress("boss-updated@domain.tld")))),
-        bcc = Some(List(EmailAddress(Some(EmailerName("My Boss 2 (updated)")), new MailAddress("boss-updated-2@domain.tld")))),
-        textSignature = TextSignature("text 2 signature"),
-        htmlSignature = HtmlSignature("html 2 signature"),
-        mayDelete = MayDeleteIdentity(true)))
-  }
+  z
```
Comment (chibenwa) [inline server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/identity/CustomIdentityDAOContract.scala:159]: oups `z` ?

File: server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/identity/IdentityRepositoryTest.scala [hunk 2]
```diff
@@ -197,4 +215,17 @@ class IdentityRepositoryTest {
         htmlSignature = HtmlSignature("html 2 signature"),
         mayDelete = MayDeleteIdentity(true)))
   }
+
+  @Test
+  def updateShouldFailWhenEmailNotExists(): Unit = {
```
Comment (quantranhong1999) [inline server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/identity/IdentityRepositoryTest.scala:220]: whenAliasNotExists maybe?

File: server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/identity/IdentityRepositoryTest.scala [hunk 3]
```diff
@@ -132,37 +151,34 @@ class IdentityRepositoryTest {
   }
 
   @Test
-  def updateShouldSuccessWhenCustomExists(): Unit = {
-    when(identityFactory.listIdentities(BOB)).thenReturn(List())
-    val customIdentity: Identity = SMono.fromPublisher(customIdentityDAO.save(BOB, CREATION_REQUEST)).block()
+  def updateShouldSuccessWhenCustomExistsAndEmailExists(): Unit = {
```
Comment (quantranhong1999) [inline server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/identity/IdentityRepositoryTest.scala:154]: updateShouldSuccessWhenCustomExistsAndAliasExists maybe?

File: server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/identity/IdentityRepositoryTest.scala [hunk 4]
```diff
@@ -115,6 +115,25 @@ class IdentityRepositoryTest {
       .containsExactlyInAnyOrder(customIdentity)
   }
 
+  @Test
+  def listShouldNotReturnTheIdentityHasEmailNotExists(): Unit = {
```
Comment (quantranhong1999) [inline server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/identity/IdentityRepositoryTest.scala:119]: alias maybe?

**Comment by chibenwa** [comment]
@quantranhong1999 can you issue the fixups for your comments?

**Comment by quantranhong1999** [comment]
> @quantranhong1999 can you issue the fixups for your comments?

I don't think I can push directly to his repo. Should I move this PR to https://github.com/apache/james-project/pull/760 ?

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala [hunk 5]
```diff
@@ -1666,5 +1666,318 @@ trait IdentitySetContract {
...
+         |				"ids": ["$customIdentityId"]
+         |			}, "c1"
+         |		]
+         |
+         |	]
+         |}""".stripMargin
+
+    val response: String = `given`
+      .body(request)
+      .when
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala:1699]: indent

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala [hunk 6]
```diff
@@ -1666,5 +1666,318 @@ trait IdentitySetContract {
...
+         |			"c1"
+         |		]
+         |	]
+         |}""".stripMargin
+
+    val response: String = `given`
+      .body(request)
+      .when
+      .post
+      .`then`
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala:1769]: indent

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala [hunk 7]
```diff
@@ -1666,5 +1666,318 @@ trait IdentitySetContract {
...
+         |				"accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |				"ids": ["$serverIdentityId"]
+         |			}, "c1"
+         |		]
+         |	]
+         |}""".stripMargin
+
+    val response: String = `given`
+      .body(request)
+      .when
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala:1855]: indent

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala [hunk 8]
```diff
@@ -1666,5 +1666,318 @@ trait IdentitySetContract {
...
+      .post
+      .`then`
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala:1857]: indent

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala [hunk 9]
```diff
@@ -1666,5 +1666,318 @@ trait IdentitySetContract {
...
+           |						"textSignature": "Difference text signature",
+           |						"htmlSignature": "<p>Difference html signature</p>"
+           |					}
+           |				}
+           |			},
+           |			"c1"
+           |		]
+           |	]
+           |}""".stripMargin)
+      .when
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala:1916]: indent

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala [hunk 10]
```diff
@@ -1666,5 +1666,318 @@ trait IdentitySetContract {
...
+         |			"c1"
+         |		]
+         |	]
+         |}""".stripMargin
+
+    val response: String = `given`
+      .body(request)
+      .when
+      .post
+      .`then`
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala:1961]: indent

**Review by quantranhong1999 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Can you rebase to solve conflicts and get rid of the no review commit now that it's merged, and squash your fixups as well? :)

**Pull request merged**